src/veil/ringct/lightwallet.cpp

```diff
@@ -0,0 +1,1890 @@
+        if (!spent_in_chain) {
+            vUnspentTxes.push_back(pair.second);
+        }
+    }
+
+    // Get Amount and blinds of all spendable txes
+    std::vector<CWatchOnlyTx> vFinalTx;
+    if (!GetAmountAndBlindForUnspentTx(vFinalTx, vUnspentTxes, spend_secret, scan_secret, spend_pubkey,errorMsg)) {
+        LogPrintf("Failed - %s\n", errorMsg);
+        return errorMsg;
+    }
+
+    // Get Random outputs
+    std::set<int64_t> setIndexSelected;
+    std::vector<CLightWalletAnonOutputData> vecSelectedOutputs;
+    std::string sError;
+    GetMainWallet()->GetAnonWallet()->GetRandomHidingOutputs(11, 11, setIndexSelected,vecSelectedOutputs, errorMsg);
+
+    std::string txHex = "";
+
+    if (!BuildLightWalletTransaction(args, vFinalTx, vecSelectedOutputs, txHex, errorMsg)) {
+        LogPrintf("Failed testing transaction");
+    }
+
+    return txHex;
+}
+
+
+bool BuildLightWalletTransaction(const std::vector<std::string>& args, const std::vector<CWatchOnlyTx>& vSpendableTx, const std::vector<CLightWalletAnonOutputData>& vDummyOutputs, std::string& txHex, std::string& errorMsg)
+{
+
+    // We need to direct this to spend the correct outputs.
+    // If the vSpendableTx has Anon and Stealth outputs we will use Anon
+
+    int nAnonTxes = 0;
+    int nStealthTxes = 0;
+
+    std::vector<CWatchOnlyTx> vAnonTxes;
+    std::vector<CWatchOnlyTx> vStealthTxes;
+
+
+    for (const auto& tx : vSpendableTx) {
+        if (tx.type == CWatchOnlyTx::ANON) {
+            vAnonTxes.push_back(tx);
+            nAnonTxes++;
+        }
+
+        if (tx.type == CWatchOnlyTx::STEALTH) {
+            vStealthTxes.push_back(tx);
+            nStealthTxes++;
+        }
+    }
+
+
+    if (nAnonTxes > 0) {
+        // We use Anon / RingCT
+        return BuildLightWalletRingCTTransaction(args, vAnonTxes, vDummyOutputs, txHex, errorMsg);
+    } else if (nStealthTxes > 0) {
+        return BuildLightWalletStealthTransaction(args, vStealthTxes, txHex, errorMsg);
+    } else {
+        errorMsg = "No Anon or Stealth txes given to build transaction";
+        return false;
+    }
+
+
+}
+
+
+/** Before you build the transaction, you must of already called
+ * API - Get Transaction for the scan_secret
+ * Light Node - Get the keyimages for the transactions -> GetKeyImagesFromAPITransactions
+ * API - Check if Key Images are spent
+ * Light Node - Get Amounts and blinds of all transactions that aren't spent -> GetAmountAndBlindForUnspentTx
+ * Light Node - Check you have at least the send amount + 0.01 Veil as a fee
+ * API - Fetch Valid Dummy Outputs
+ */
+// Args include, the following
+// Spend Secret
+// Scan Secret
+// Spend Public Key
+// Address to spend to
+// Amount to spend
+bool BuildLightWalletRingCTTransaction(const std::vector<std::string>& args, const std::vector<CWatchOnlyTx>& vSpendableTx, const std::vector<CLightWalletAnonOutputData>& vDummyOutputs, std::string& txHex, std::string& errorMsg)
```

Comment:
This, plus the rest of the file, looks like a ton of copied code just to add in something lightwallet-related to each function? Is there a way we can deal with that directly in the original code instead or otherwise share functionality?